tests/unit_tests/client/proxy/http.cc

```diff
@@ -57,4 +57,166 @@ BOOST_AUTO_TEST_CASE(ok) {
+  tmp.m_Address = "kovri.i2p";
+  // Process jump service request (should return false)
+  BOOST_CHECK(!tmp.HandleJumpService());
+}
+
+BOOST_AUTO_TEST_CASE(JumpServiceNoBase64)
+{
+  kovri::client::HTTPMessage tmp;
+  // Jump service helper with no base64-destination
+  std::string tmp_data = "GET stats.i2p?i2paddresshelper= HTTP/1.1\r\n\r\n";
+  // Load variables into HTTPMessage
+  BOOST_CHECK(tmp.HandleData(tmp_data));
+  // Mock m_Address for test output
+  tmp.m_Address = "kovri.i2p";
+  // Process jump service request (should return false)
+  BOOST_CHECK(!tmp.HandleJumpService());
+}
+
+BOOST_AUTO_TEST_CASE(CreateHTTPRequestJumpService)
+{
+  kovri::client::HTTPMessage tmp;
+  std::string tmp_data = "GET stats.i2p";
+  // Jump service GET parameter
+  tmp_data += "?i2paddresshelper=0UVPqAA4xUSfPYPBca24h8fdokhwcJZ-5OsBYvK7byXtXT~fOV2pExi8vrkgarGTNDfJbB2KCsdVS3V7qwtTvoCGYyklcDBlJsWMj7H763hEz5rt9SzLkcpwhOjXL1UB-QW8KxM30t-ZOfPc6OiJ1QpnE6Bo5OUm6jPurQGXdWCAPio5Z-YnRL46n0IHWOQPYYSSt-S75rMIKbZbEMDraRvSzYAphUaHfvtWr2rCSPkKh3EbrOiBYiAP2oWvAQCsjouPgVBbiAezHedM2gXzkgIyCV2kGOOcHhiihd~7fWwJOloH-gO78QkmCuY-3kp3633v3MB-XNKWnATZOuf2syWVBZbTnOXsWf41tu6a33HOuNsMxAOUrwbu7Q-EITwNlKN6~yZm4RKsJUsBGfVtKl8PBMak3flQAg95oV0OBDGuizIQ9vREOWvPGlQCAXZzEg~cUNbfBQAEAAcAAA%3D%3D";
+  tmp_data += " HTTP/1.1\r\n\r\n";
+  // Load variables into HTTPMessage
+  BOOST_CHECK(tmp.HandleData(tmp_data));
+  // Mock m_Address for test output
+  tmp.m_Address = "kovri.i2p";
+  // Process jump service request
+  BOOST_CHECK(tmp.CreateHTTPRequest());
+}
+
+BOOST_AUTO_TEST_CASE(CreateHTTPRequestJumpServiceInvalid)
+{
+  kovri::client::HTTPMessage tmp;
+  std::string tmp_data = "GET stats.i2p";
+  // Invalid jump service GET parameter
+  tmp_data += "?i2paddresshelpXX=0UVPqAA4xUSfPYPBca24h8fdokhwcJZ-5OsBYvK7byXtXT~fOV2pExi8vrkgarGTNDfJbB2KCsdVS3V7qwtTvoCGYyklcDBlJsWMj7H763hEz5rt9SzLkcpwhOjXL1UB-QW8KxM30t-ZOfPc6OiJ1QpnE6Bo5OUm6jPurQGXdWCAPio5Z-YnRL46n0IHWOQPYYSSt-S75rMIKbZbEMDraRvSzYAphUaHfvtWr2rCSPkKh3EbrOiBYiAP2oWvAQCsjouPgVBbiAezHedM2gXzkgIyCV2kGOOcHhiihd~7fWwJOloH-gO78QkmCuY-3kp3633v3MB-XNKWnATZOuf2syWVBZbTnOXsWf41tu6a33HOuNsMxAOUrwbu7Q-EITwNlKN6~yZm4RKsJUsBGfVtKl8PBMak3flQAg95oV0OBDGuizIQ9vREOWvPGlQCAXZzEg~cUNbfBQAEAAcAAA%3D%3D";
+  tmp_data += " HTTP/1.1\r\n\r\n";
+  // Load variables into HTTPMessage
+  BOOST_CHECK(tmp.HandleData(tmp_data));
+  // Mock m_Address for test output
+  tmp.m_Address = "kovri.i2p";
+  // Process non-jump service request
+  BOOST_CHECK(tmp.CreateHTTPRequest());
+}
+
+BOOST_AUTO_TEST_CASE(CreateHTTPRequestJumpServiceNoBase64)
+{
+  kovri::client::HTTPMessage tmp;
+  // Valid jump service key with no base64 destination
+  std::string tmp_data = "GET stats.i2p?i2paddresshelper= HTTP/1.1\r\n\r\n";
+  // Load variables into HTTPMessage
+  BOOST_CHECK(tmp.HandleData(tmp_data));
+  // Mock m_Address for test output
+  tmp.m_Address = "kovri.i2p";
+  // Process invalid jump service request
+  BOOST_CHECK(!tmp.CreateHTTPRequest());
+}
```

Comment:
>kovri::client::HTTPMessage tmp;
>std::string tmp_data = "GET stats.i2p";
>tmp.m_Address = "kovri.i2p"; 

All this copypasta within the testcases points out a glaring issue: you need to use a test fixture! See other tests for an example and/or read up on boost.test.

Reply:
Entirely, will fix with a fixture :)